vars/rollingUpgradePipeline.groovy

```diff
@@ -99,17 +99,22 @@ def call(Map pipelineParams) {
                                 wrap([$class: 'BuildUser']) {
                                     dir('scylla-cluster-tests') {
                                         checkout scm
-                                        ArrayList base_versions_list = params.base_versions.contains('.') ? params.base_versions.split('\\,') : []
-                                        supportedVersions = supportedUpgradeFromVersions(
-                                            base_versions_list,
-                                            pipelineParams.linux_distro,
-                                            params.new_scylla_repo
-                                        )
-                                        (testDuration,
-                                         testRunTimeout,
-                                         runnerTimeout,
-                                         collectLogsTimeout,
-                                         resourceCleanupTimeout) = getJobTimeouts(params, builder.region)
+                                        try {
+                                            ArrayList base_versions_list = params.base_versions.contains('.') ? params.base_versions.split('\\,') : []
+                                            supportedVersions = supportedUpgradeFromVersions(
+                                                base_versions_list,
+                                                pipelineParams.linux_distro,
+                                                params.new_scylla_repo
+                                            )
+                                        } catch (Exception e) {
+                                            println(e)
+                                            error("Run aborted due to inability to list supported upgrade-from versions.")
```

Comment:
as i'm not very familiar with groovy, is this the way to "stop" the pipeline, with error status?

Reply:
I'm not a Groovy specialist myself. Calling error would do the trick, but the easier way turned out to be simply removing the `catchError` wrapper.  Which I did. Results with a forced error of the hydra script here: https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/maciej/job/mc-rolling-upgrade-centos-07-base-version-fix/18

Reply:
build 18 failed with something different:
```
[](https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/maciej/job/mc-rolling-upgrade-centos-07-base-version-fix/18/console#)11:37:56  [Pipeline] sh
11:37:58  + ./docker/env/hydra.sh get-scylla-base-versions --only-print-versions true --linux-distro centos --scylla-re
11:38:08  Usage: sct.py get-scylla-base-versions [OPTIONS]
11:38:08  Try 'sct.py get-scylla-base-versions --help' for help.
11:38:08  
11:38:08  Error: No such option: --scylla-re (Possible options: --scylla-repo, --scylla-version)
11:38:09  Running on Build Server...
11:38:09  Docker version 20.10.12, build e91ed57
11:38:09  Pull version v1.10 from Docker Hub...
11:38:09  v1.10: Pulling from scylladb/hydra
11:38:09  Digest: sha256:b594c252923fca64ac2d069aa8b70f6ef19b3d60325832f2591d017e74514952
11:38:09  Status: Image is up to date for scylladb/hydra:v1.10
11:38:09  docker.io/scylladb/hydra:v1.10
11:38:09  Obtaining QA SSH keys...
11:38:09  QA SSH keys obtained.
11:38:09  Making sure the ownerships of results directories are of the user
11:38:09  Going to run './sct.py  get-scylla-base-versions --only-print-versions true --linux-distro centos --scylla-re'...
```